### PR TITLE
Feat/refactor densityplot script

### PR DESCRIPTION
**Pull request by ashwini06**
### This PR: 
Refators rule `seaborn_densityplot_umi`.

- Moved chunk of python script to a standalone script

- Added tests to check the related density plot functions

Fixed: 
- Bug in the rule `bcftools_query_calculatenoiseAF_umi`
- Missing '%' symbol in the bcftools query 

### Review and tests: 
- [ ] Tests pass
- [ ] Code review
- [ ] New code is executed and covered by tests, and test approve

### CODE REVIEW

File: BALSAMIC/utils/workflowscripts.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import pandas as pd
+import seaborn as sns
+import matplotlib.pyplot as plt
+
+def get_file_contents(input_file, prefix_name):
+    """ Reads the 2-column tsv file and returns file contents with header names  """
```
Comment (hassanfa) [inline BALSAMIC/utils/workflowscripts.py:6]: Please expand docstring here. similar to what you did for models.

File: BALSAMIC/utils/workflowscripts.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    file = pd.read_csv(input_file, sep='\t', header=None)
+    file.columns = ['id', 'AF']
+    file['method'] = prefix_name
+    return(file)
```
Comment (hassanfa) [inline BALSAMIC/utils/workflowscripts.py:10]: omit `()` if returning only one value.

File: BALSAMIC/utils/workflowscripts.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+def get_densityplot(input_file1, input_file2, prefix_name1, prefix_name2, output_pdf):
+    """ Reads two input dataframes and outputs a densityplot to the pdf file """
```
Comment (hassanfa) [inline BALSAMIC/utils/workflowscripts.py:13]: Update Docstring here as well please.

File: BALSAMIC/utils/workflowscripts.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    dataframe1 = get_file_contents(input_file1,prefix_name1)
+    dataframe2 = get_file_contents(input_file2, prefix_name2)
+    sns.kdeplot(dataframe1['AF'], color='r', shade=True, label=set(dataframe1['method']))
+    sns.kdeplot(dataframe2['AF'], color='b', shade=True, label=set(dataframe2['method']))
+    plt.xlabel('Allelic Frequency (AF)')
+    plt.ylabel('Probability Density')
+    return(plt.savefig(output_pdf))
```
Comment (hassanfa) [inline BALSAMIC/utils/workflowscripts.py:20]: similar to above, omit `()` if returning only one value.

File: requirements.txt [hunk 5]
```diff
@@ -15,5 +15,5 @@ pydantic
 numpy
 cyvcf2<0.10.0
 pandas>1.1.0
-seaborn>=0.11.0
-matplotlib>=3.3.0
+seaborn==0.10.1
+matplotlib>=3.3.2
```
Comment (hassanfa) [inline requirements.txt:19]: seaborn or matplotlib are already in dep list. Could you cross check please? Also AB-order them.

**Review by hassanfa — APPROVED** [review]
minor stuff.

By the way, could you format your files with yapf please?

`yapf -i path_to_file.py`

File: BALSAMIC/utils/workflowscripts.py (hunk 1 above)
Comment (ashwini06) [reply BALSAMIC/utils/workflowscripts.py]: Fixed!

File: BALSAMIC/utils/workflowscripts.py (hunk 3 above)
Comment (ashwini06) [reply BALSAMIC/utils/workflowscripts.py]: Fixed

File: BALSAMIC/utils/workflowscripts.py (hunk 2 above)
Comment (ashwini06) [reply BALSAMIC/utils/workflowscripts.py]: Fixed

File: requirements.txt (hunk 5 above)
Comment (ashwini06) [reply requirements.txt]: I locked the seaborn version to 0.10.1 for some specific reasons for plotting densityplots.

**Pull request merged**